src/api/spec/factories/attribs.rb

```diff
@@ -7,11 +7,16 @@
       package nil
     end
 
+    before(:create) do |attrib, _evaluator|
+      User.current = User.find_by_login!(attrib.creator)
+    end
+
     after(:create) do |attrib, evaluator|
       if evaluator.package
         attrib.package = evaluator.package
         attrib.project = nil
       end
+      User.current = nil
```

Comment:
Why shouldn't this be valid?

```ruby
RSpec.describe BLAH, type: :model do
  let(:attribute) { create(:attrib) }
  let(:user] { create(:confirmed_user) }

  describe '#whatever' do
    login(user)
    it { expect(attribute.fullname).to eq("#{attribute.namespace}:#{attribute.name}") }
  end
end
```

Reply:
the problem is that the create(:attrib) in your 2nd line will already crash. I changed this now to build

Reply:
How would it crash if User.current is set?

Reply:
I didn't spot the difference between let and let!

Reply:
Happened to all of us